sympy/concrete/products.py

```diff
@@ -201,18 +201,23 @@ def term(self):
     function = term
 
     def _eval_is_zero(self):
-        # a Product is zero only if its term is zero.
-        return self.term.is_zero
+        if self.has_finite_limits:
+            # a Product is zero only if its term is zero.
+            return self.term.is_zero
 
     def _eval_is_extended_real(self):
         return self.function.is_extended_real
 
     def _eval_is_positive(self):
-        if self.function.is_positive:
+        if self.function.is_positive and self.has_finite_limits:
             return True
 
     def _eval_is_nonnegative(self):
-        if self.function.is_nonnegative:
+        if self.function.is_nonnegative and self.has_finite_limits:
+            return True
```

Comment:
An `_eval_is_finite` method might be useful. The `_eval_is_nonnegative` method above for example could be automatically inferred from the extended version below if you had that. Similarly if you had `_eval_is_zero` then `_eval_is_positive` could be inferred.

Reply:
Isn't that basically the same as `is_convergent`? I was considering it, but since it is quite computationally intensive, I didn't really want to involve it into the random evaluation of assumptions...

`is_zero` could just check for the argument being zero. Everything else require some sort of evaluation. So not sure how much it would help.

Reply:
I would expect an `_eval_is_finite` method to be a quick check like: finite limits and finite summand -> return True, otherwise return None. This is essentially the same check that you have used to determine finiteness here.

I think there should be something like is_convergent but I don't think it should be a property. It should be a method with a docstring explaining that it is slow and potentially with options to control which algorithm is used etc.

Reply:
`is_zero` would just check whether the argument is zero and the finiteness of the limits. It helps because you don't need to define all the `_eval_is` methods. Positive can be inferred from nonnegative and zero. In combination if you define extended_nonnegative, finite and zero then they can be used to infer all of positive, nonnegative, extended_positive etc. Also finite and zero are the most directly useful in core operations like division.

Reply:
I think that some are easier than others to define. For `is_finite` one will need to check that there are no singularities in the range and that the bounds are finite. First one is not so simple for the general case. Hence, providing those that are reasonably easy to evaluate may be a better/working (initial) strategy as opposed to putting much effort into solving a potentially better check, taking more time.

Reply:
I would assume that the assumption system continues checking other properties when `None` is obtained?

Reply:
If the dummy symbols have assumptions that are consistent with the limits then function.is_finite and has_finite_limits are enough to imply that a Sum/Product is finite I think.

Reply:
I see your point. That would indeed be a way to check it.